### PR TITLE
AP_Frsky_Telem: fix for protocol=4 GAlt=0 and GSPd=0

### PR DESCRIPTION
**Pull request by yaapu**
This fixes serial_protocol 4 GPS altitude and speed error, both were reported as constant 0
for calc_gps_position() was never called.
Issue introduced after adding support for VSpd and trying to execute "slow" code when there was no serial input to process.

### CODE REVIEW

**Comment by peterbarker** [comment]
Would be nice to fill in some of the validators here: https://github.com/ardupilot/ardupilot/blob/master/Tools/autotest/common.py#L4850

Will catch some regressions, at least.

**Comment by yaapu** [comment]
something like  tfp_validate_vel_and_yaw(self, value)?

File: libraries/AP_Frsky_Telem/AP_Frsky_Telem.h [hunk 1]
```diff
@@ -206,11 +206,12 @@ class AP_Frsky_Telem {
     struct
     {
         bool sport_status;
+        bool gps_refresh = false;
```
Comment (tridge) [inline libraries/AP_Frsky_Telem/AP_Frsky_Telem.h]: we don't need the init of these vars, it just wastes flash

Comment (yaapu) [reply libraries/AP_Frsky_Telem/AP_Frsky_Telem.h]: ok bool initialization fixed

**Comment by peterbarker** [comment]
@yaapu could you rebase the testing PR, please?

**Pull request merged**